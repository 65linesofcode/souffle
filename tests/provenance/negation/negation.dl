// Souffle - A Datalog Compiler
// Copyright (c) 2017, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// This code tests the provenance explain interface for a simple path example.

<<<<<<< HEAD
.pragma "provenance" " "
=======
.pragma "provenance" "2"
>>>>>>> 5fd84e72e8c97f18f5c71d9db6edfb3ae378c9e0

.decl person(x:symbol)
person("john").
person("jim").
person("tim").

.decl being(x:symbol)
being(x) :- person(x).
being("tomothy_the_cat").
being("ollie_the_doggo").

.decl animal(x:symbol)
animal(x) :- being(x), !person(x).
.output animal
