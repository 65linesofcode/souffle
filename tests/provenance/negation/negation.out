Explain is invoked.
Enter command > animal
being("ollie_the_doggo") !person("ollie_the_doggo") 
------------------------------------------------(R1)
             animal("ollie_the_doggo")              
Enter command > Exiting explain
